src/main/java/seedu/loyaltylift/storage/JsonAdaptedOrder.java

```diff
@@ -0,0 +1,110 @@
+
+import seedu.loyaltylift.commons.exceptions.IllegalValueException;
+import seedu.loyaltylift.model.attribute.Address;
+import seedu.loyaltylift.model.attribute.Name;
+import seedu.loyaltylift.model.order.CreatedDate;
+import seedu.loyaltylift.model.order.Order;
+import seedu.loyaltylift.model.order.Quantity;
+import seedu.loyaltylift.model.order.Status;
+
+/**
+ * Jackson-friendly version of {@link Order}.
+ */
+public class JsonAdaptedOrder {
+
+    public static final String MISSING_FIELD_MESSAGE_FORMAT = "Order's %s field is missing!";
+
+    private final String name;
+    private final Integer quantity;
+    private final String status;
+    private final String address;
+    private final String createdDate;
+
+    /**
+     * Constructs a {@code JsonAdaptedOrder} with the given order details.
+     */
+    @JsonCreator
+    public JsonAdaptedOrder(@JsonProperty("name") String name, @JsonProperty("phone") Integer quantity,
+                            @JsonProperty("email") String status, @JsonProperty("address") String address,
+                            @JsonProperty("createdDate") String createdDate) {
+        this.name = name;
+        this.quantity = quantity;
+        this.status = status;
+        this.address = address;
+        this.createdDate = createdDate;
+    }
+
+    /**
+     * Converts a given {@code Order} into this class for Jackson use.
+     */
+    public JsonAdaptedOrder(Order source) {
+        name = source.getName().fullName;
+        quantity = source.getQuantity().value;
+        status = source.getStatus().toString().toUpperCase();
+        address = source.getAddress().value;
+        createdDate = source.getCreatedDate().toString();
+    }
+
+    /**
+     * Converts this Jackson-friendly adapted customer object into the model's {@code Order} object.
+     *
+     * @throws IllegalValueException if there were any data constraints violated in the adapted order.
+     */
+    public Order toModelType() throws IllegalValueException {
+        if (name == null) {
+            throw new IllegalValueException(String.format(MISSING_FIELD_MESSAGE_FORMAT, Name.class.getSimpleName()));
+        }
+        if (!Name.isValidName(name)) {
+            throw new IllegalValueException(Name.MESSAGE_CONSTRAINTS);
+        }
+        final Name modelName = new Name(name);
+
+        if (quantity == null) {
+            throw new IllegalValueException(
+                    String.format(MISSING_FIELD_MESSAGE_FORMAT, Quantity.class.getSimpleName()));
+        }
+        if (!Quantity.isValidQuantity(quantity)) {
+            throw new IllegalValueException(Quantity.MESSAGE_CONSTRAINTS);
+        }
+        final Quantity modelQuantity = new Quantity(quantity);
+
+        if (status == null) {
+            throw new IllegalValueException(String.format(MISSING_FIELD_MESSAGE_FORMAT, Status.class.getSimpleName()));
+        }
+        final Status modelStatus = Status.valueOf(status);
```

Comment:
missing isValidStatus check? o_O
isValidStatus method in Status doesnt seem to be currently present though

Reply:
I believe this is because `Status` is an enum. If `status` is not null, then it must be one of the constants of `Status` enums, hence it's always valid.

Reply:
oh thats true